opencti-platform/opencti-graphql/src/domain/enrichment.js

```diff
@@ -40,3 +46,35 @@ export const createEntityAutoEnrichment = async (context, user, element, scope)
   );
   return workList;
 };
+
+const findConnectorsForElementEnrichment = async (context, user, element, scope) => {
+  const connectors = await getEntitiesListFromCache(context, user, ENTITY_TYPE_CONNECTOR);
+  return filterConnectorsForElementEnrichment(context, connectors, element, scope);
+};
+
+export const filterConnectorsForElementEnrichment = async (context, connectors, element, scope) => {
+  // first filter active & enrichment connectors only
+  const activeConnectors = connectors.filter((conn) => conn.active === true && conn.connector_type === CONNECTOR_INTERNAL_ENRICHMENT);
+  const targetConnectors = [];
+  for (let i = 0; i < activeConnectors.length; i += 1) {
+    const conn = activeConnectors[i];
+    const scopeMatch = scope ? (conn.connector_scope ?? []).some((s) => s.toLowerCase() === scope.toLowerCase()) : true;
+    const autoTrigger = conn.connector_trigger_filters ? await isStixMatchConnectorFilter(context, element, conn.connector_trigger_filters) : conn.auto === true;
+    if (scopeMatch && autoTrigger) {
+      targetConnectors.push(conn);
+    }
+  }
+  return targetConnectors;
+};
+
+const isStixMatchConnectorFilter = async (context, element, stringFilters) => {
+  if (!stringFilters) {
+    return true; // no filters, match all
+  }
+  const jsonFilters = JSON.parse(stringFilters);
+  if (!isFilterGroupNotEmpty(jsonFilters)) {
+    return false; // filters empty -> we don't try to match
+  }
```

Comment:
Is is intended that no filters means matching all, but filters empty means matching nothing?

Reply:
oh right, I changed the logic, and I forgot about this part

Reply:
done